Automattic/jetpack
# VideoPress: Load full media item details from backend when uploading videos to VideoPress

The media item that is added to the Attachments collection of the Media Library immediately after a video upload to VideoPress is completed is a half-populated js object that contained the minimal amount of fields to display the media item, but was missing the VideoPress "compat" fields. A full page refresh was needed to reload the Media Library in order for the media item to display all of the expected fields.

This PR uses the existing backbone.js api to request the single attachment from the backend and update the media Attachments collection (which automatically updates the UI) so that all expected fields are available immediately after upload.

**Before**
<img width="467" alt="Screen Shot 2021-11-02 at 1 47 02 PM" src="https://user-images.githubusercontent.com/4081020/139918460-dd7e9c78-8235-450d-8a92-e63fa165d5af.png">


**After**
<img width="475" alt="Screen Shot 2021-11-02 at 1 44 37 PM" src="https://user-images.githubusercontent.com/4081020/139918035-b64e761a-1d29-47e4-ae2c-eec7039c8670.png">

Fixes # 984-gh-Automattic/greenhouse

#### Changes proposed in this Pull Request:

* Do an extra request to the backend after wpcom upload completes to pull the actual Attachment data from Jetpack/wp.org site

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


Nope

#### Testing instructions:






* run `jetpack build plugins/jetpack`
* For a site that has VideoPress enabled, go to the Media Library `wp-admin/upload.php`
* Upload a new video
* Once the file has finished uploading, click the new Media Item. The media item view should include VideoPress "compat" fields at the bottom of the right side bar. Previously, these were missing until after the page was refreshed and all data was re-loaded.
